libcst/metadata/scope_provider.py

```diff
@@ -698,10 +700,19 @@ def visit_Param(self, node: cst.Param) -> Optional[bool]:
                 attr.visit(self)
         return False
 
-    def visit_Arg_keyword(self, node: cst.Arg) -> None:
-        keyword = node.keyword
-        if keyword is not None:
-            self.scope.record_assignment(keyword.value, node)
+    def visit_Arg(self, node: cst.Arg) -> Optional[bool]:
+        # The keyword of Arg is neither an Assignment nor an Access and we explicitly don't visit it.
+        for attr in [
+            node.value,
+            node.equal,
+            node.comma,
+            node.star,
+            node.whitespace_after_star,
+            node.whitespace_after_arg,
+        ]:
```

Comment:
I don't really understand why this visits everything other than `node.value`, there can't possibly be assignments and accesses in the rest of them. But then again, the rest of the methods in this class do the same so 🤷

Reply:
Correct, we don't expect to find Name in those attributes.
Initially, when ScopeProvide was implemented, the goal was to provide a scope as a container for all nodes in the syntax tree. There is a trick in on_leave that we set the scope object to a node automatically when it's visited and on leave.
https://github.com/Instagram/LibCST/blob/8c77811d701e8c8d58a82f62acd8e08d9bfd7048/libcst/metadata/scope_provider.py#L846-L847
By returning a False in a visit function, we stop the traversal of a subtree and that'll cause some node doesn't have a scope associated.

When we look back to review this design, it probably isn't necessary to provide a scope for a formatting node (e.g. comma, white space, comment.) The use cases are most likely just need the scope object from a non-formatting node. I can add a TODO here to clean up scope node associates to formatting for follow up task.